public/js/domCons.js

```diff
@@ -0,0 +1,102 @@
+const contentNode = document.getElementById('content');
+const mainMessage = document.getElementById('mainMessage');
+mainMessage.className = 'main-message';
+
+const displayData = (data) => {
+  data.forEach((element) => {
+    const newNode = document.createElement('div');
```

Comment:
@AymanAlqoqa Another approach you could use, to build this UI of questions, is to put an invisible question "template" into your html, and then clone it several times here inside the displayData loop.

So first you'd put a "template" of your question html into consProfile.hbs somewhere, and make it invisible so the user doesn't see it:
```
<div style='display: none' id='question-item-template'>
  <div class='titleNode--title'><span class='container__content--title'></span></div>
  ...
  ...
```
Next, here inside the forEach loop you'd clone that element (and make your cloned node visible):
```
const newNode = document.querySelector('#question-item-template').cloneNode(true);
newNode.style.display = 'block';
```
And then you'd add text and event listeners:
```
newNode.querySelector('.container__content--title').textContent = `<<< ${element.title} >>>`;
newNode.querySelector('.submitBtn').addEventListener(....
```

That might be easier to understand (and maintain, and make updates) to in the future compared to building the dom structure out of many createElement and appendChild calls.

Reply:
great approach, thanks